src/libAtomVM/nifs.h

```diff
@@ -48,6 +48,19 @@ extern "C" {
 
 const struct Nif *nifs_get(AtomString module, AtomString function, int arity);
 
+/**
+ * @brief Convenient function to return posix errors as atom.
+ *
+ * @details POSIX does not define the values of errno errors, so this function
+ * makes sure Erlang code can interpret error codes whatever the platform.
+ *
+ * @param err the error code, typically obtained from `errno(3)`
+ * @param glb the global context
+ * @return an atom representing the error or `UNDEFINED_ATOM` if the error
+ * number is not known.
+ */
+term posix_errno_to_atom(int err, GlobalContext *glb);
```

Comment:
I have an open question here, that I would like to discuss with you before moving code around: do we really want to keep adding POSIX specific NIFs to the general nifs file, or do we want to create a `posix_nifs.h` or anything like that file?
Let's discuss this change.